plaso/parsers/setupapi.py

```diff
@@ -0,0 +1,205 @@
+
+  _SLASH = pyparsing.Literal('/').suppress()
+
+  _FOUR_DIGITS = text_parser.PyparsingConstants.FOUR_DIGITS
+  _TWO_DIGITS = text_parser.PyparsingConstants.TWO_DIGITS
+
+  _SETUPAPI_DATE_TIME = pyparsing.Group(
+      _FOUR_DIGITS.setResultsName('year') + _SLASH +
+      _TWO_DIGITS.setResultsName('month') + _SLASH +
+      _TWO_DIGITS.setResultsName('day') +
+      text_parser.PyparsingConstants.TIME_MSEC_ELEMENTS
+  )
+
+  _SETUPAPI_LINE = (
+      pyparsing.SkipTo('>>>  [', include=True).suppress() +
+      pyparsing.SkipTo(']').setResultsName('entry_type') +
+      pyparsing.SkipTo('>>>  Section start', include=True).suppress() +
+      _SETUPAPI_DATE_TIME.setResultsName('start_time') +
+      pyparsing.SkipTo('<<<  Section end ').setResultsName('message') +
+      # _SETUPAPI_DATE_TIME.setResultsName('end_time') +
+      pyparsing.SkipTo('<<<  [Exit status: ', include=True).suppress() +
+      pyparsing.SkipTo(']').setResultsName('exit_status') +
+      pyparsing.SkipTo(pyparsing.lineEnd()) +
+      pyparsing.ZeroOrMore(pyparsing.lineEnd()))
+
+  LINE_STRUCTURES = [
+      ('logline', _SETUPAPI_LINE),
+  ]
+
+  def _GetISO8601String(self, structure):
+    """Retrieves an ISO 8601 date time string from the structure.
+
+    Args:
+      structure (pyparsing.ParseResults): structure of tokens derived from a
+          log entry.
+
+    Returns:
+      str: ISO 8601 date time string.
+
+    Raises:
+      ValueError: if the structure cannot be converted into a date time string.
+    """
+    year = self._GetValueFromStructure(structure, 'year')
+    month = self._GetValueFromStructure(structure, 'month')
+    day_of_month = self._GetValueFromStructure(structure, 'day')
+    hours = self._GetValueFromStructure(structure, 'hours')
+    minutes = self._GetValueFromStructure(structure, 'minutes')
+    seconds = self._GetValueFromStructure(structure, 'seconds')
+    microseconds = self._GetValueFromStructure(structure, 'microseconds')
+
+    try:
+      iso8601 = (
+          '{0:04d}-{1:02d}-{2:02d}T{3:02d}:{4:02d}:{5:02d}.{6:03d}').format(
+              year, month, day_of_month, hours, minutes, seconds, microseconds)
+    except (TypeError, ValueError) as exception:
+      raise ValueError(
+          'unable to format date time string with error: {0!s}.'.format(
+              exception))
+
+    return iso8601
+
+  def _ParseRecordLogline(self, parser_mediator, structure):
+    """Parses a logline record structure and produces events.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          log entry.
+    """
+    date_time = dfdatetime_time_elements.TimeElementsInMilliseconds()
+
+    time_elements_structure = self._GetValueFromStructure(
+        structure, 'start_time')
+    try:
+      datetime_iso8601 = self._GetISO8601String(time_elements_structure)
```

Comment:
Consider replacing `_GetISO8601String` by:

```
TimeElementsInMilliseconds(time_elements_tuple=time_elements_structure)
```
You might need to tweak the `_SETUPAPI_DATE_TIME` definition a bit to return a group (as a tuple)